src/openrct2/drawing/Sprite.cpp

```diff
@@ -36,6 +36,75 @@
 using namespace OpenRCT2;
 using namespace OpenRCT2::Ui;
 
+static inline uint32 rctc_to_rct2_index(uint32 image) {
+    if      (image < 1575                   ) return image;
+    else if (image >=  1607 && image <  4983) return image - 32;
+    else if (image >=  4986 && image < 17189) return image - 35;
+    else if (image >= 17191 && image < 18121) return image - 37;
+    else if (image >= 18123 && image < 23800) return image - 39;
+    else if (image >= 23804 && image < 24670) return image - 43;
+    else if (image >= 24674 && image < 28244) return image - 47;
+    else if (image >= 28246                 ) return image - 49;
+    else throw Exception(); // invalid RCT2 element
+}
+
+static void read_and_convert_gxdat(IStream * stream, size_t count, bool is_rctc, rct_g1_element *elements)
+{
+    auto g1Elements32 = stream->ReadArray<rct_g1_element_32bit>(count);
+    if (is_rctc) // RCTC element count
+    {
+        uint32 rctc = 0;
+        for (size_t i = 0; i < 29294; ++i)
+        {
+            if (i == 1575) rctc += 32;
+            else if (i == 23761 || i == 24627) rctc += 4;
+            else if (i == 4951) rctc += 3;
+            else if (i == 17154 || i == 18084 || i == 28197) rctc += 2;
+
+            const rct_g1_element_32bit &src = g1Elements32[rctc];
+
+            /* Double cast to silence compiler warning about casting to
+             * pointer from integer of mismatched length.
+             */
+            elements[i].offset        = (uint8*)(uintptr_t)src.offset;
```

Comment:
`(uint8 *)`

Reply:
This was like this before I got to it 😜. According to the comment, it's to avoid a compiler warning. Apparently, converting a 32-bit integer to a 64-bit pointer produces a warning on one of our supported compilers. (not that I'd know which one). If this is safe to change, I will, but I'm assuming it will only break the CI builds at this point.